core/modules/node/node.pages.inc

```diff
@@ -40,8 +40,16 @@ function node_add_page() {
       );
     }
   }
-  // Bypass the node/add listing if only one content type is available.
-  if (count($content) == 1) {
+  // If there are more than one content types, sort them by title instead of
+  // machine name before displaying the list of links...
+  if (count($content) !== 1) {
```

Comment:
can we have zero content types?  I've never tried to delete all the content types.

if we can have zero content types we could change the `else` to `elseif (count($content) == 1)` which should provide the enhancement and keep parity with existing functionality.